packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -177,18 +184,23 @@ function UploadBlankSlate(props: {
   );
 }
 
-function mockApiCall(image: File) {
-  return new Promise<Slice>((resolve) => {
+function mockUpload(_image: File) {
+  return new Promise<
+    { imageUrl: string; status: "success" } | { status: "error" }
+  >((resolve) => {
     setTimeout(
       () => {
-        resolve({
-          thumbnailUrl:
-            "https://images.unsplash.com/photo-1588315029754-2dd089d39a1a?w=512",
-          image,
-          status: "success",
-        });
+        resolve(
+          Math.random() > 0.2
+            ? {
+                imageUrl:
+                  "https://images.unsplash.com/photo-1588315029754-2dd089d39a1a?w=512",
+                status: "success",
+              }
+            : { status: "error" },
```

Comment:
not important because we're going to remove this, but why not reject directly here so you don't have to do it in the caller side?

Reply:
I did it like this because I figured it would better simulate what the actual API calls would resemble in the actual version. If that's not the case, then I'll update it when we know how it will actually work.

Reply:
sounds good